From 1e51875c7b2936c224deab07d6f4b757b0d959c1 Mon Sep 17 00:00:00 2001
From: Xiao Chen <xiao@apache.org>
Date: Tue, 15 Nov 2016 13:54:51 -0800
Subject: [PATCH 2512/2748] =?UTF-8?q?YARN-5875.=20TestTokenClientRMService#t?=
 =?UTF-8?q?estTokenRenewalWrongUser=20fails.=20Contributed=20by=20Gergely=20?=
 =?UTF-8?q?Nov=C3=A1k.?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

(cherry picked from commit d5c396ccb361e1a0cf4307ce1d1b37fae0eec250)
(cherry picked from commit fc4c0595be8b1188df07c9313434327447487919)

 Conflicts:
	hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/TestTokenClientRMService.java

Change-Id: I34eb51213cd05c09daeddc489011f94b64805405
---
 .../resourcemanager/TestClientRMService.java       |    7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/TestClientRMService.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/TestClientRMService.java
index 59010f6..6bceeb2 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/TestClientRMService.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/TestClientRMService.java
@@ -612,9 +612,10 @@ public Void run() throws Exception {
             checkTokenRenewal(owner, other);
             return null;
           } catch (YarnException ex) {
-            Assert.assertTrue(ex.getMessage().contains(owner.getUserName() +
-                " tries to renew a token with renewer " +
-                other.getUserName()));
+            Assert.assertTrue(ex.getMessage().contains(
+                owner.getUserName() + " tries to renew a token"));
+            Assert.assertTrue(ex.getMessage().contains(
+                "with non-matching renewer " + other.getUserName()));
             throw ex;
           }
         }
-- 
1.7.9.5

